File: 프로그래머스/lv2/131704. 택배상자/택배상자.go
//import "fmt"
func solution(orders []int) int {
    var result int = len(orders)
    var sub_belt []int
    var cargo int = 1
    
    for index, order := range orders {
        if order > cargo {
            for ; cargo < order; cargo++ {
                sub_belt = append(sub_belt, cargo)
            }
            cargo++
        } else if order == cargo {
            cargo++
        } else {
            if len(sub_belt) == 0 || sub_belt[len(sub_belt)-1] != order {
                result = index
                break
            }            
            sub_belt = sub_belt[:len(sub_belt)-1]            
        }
    } 
    
    return result
}

File: 프로그래머스/lv2/131704. 택배상자/택배상자.py
from collections import deque
def solution(order):
    m = len(order)
    order = deque(order)
    stacks = deque()
    boxs = deque([i for i in range(1,len(order)+1)])
    while order :
        if boxs :
            if boxs[0] == order[0] :
                boxs.popleft()
                order.popleft()
                continue
        if stacks :
            if stacks[-1] == order[0] :
                stacks.pop()
                order.popleft()
                continue
        
        if boxs :
            stacks.append(boxs.popleft())
        else :
            break
        
        
    return m - len(order)